htdocs/holiday/class/holiday.class.php

```diff
@@ -1132,45 +1132,30 @@ function updateSoldeCP($userID='',$nbHoliday='', $fk_type='')
 				$result = $this->db->query($sql);
 
 				$typeleaves=$this->getTypes(1,1);
-				foreach($typeleaves as $key => $val)
-				{
-					// On ajoute x jours à chaque utilisateurs
-					$nb_holiday = $val['newByMonth'];
-					if (empty($nb_holiday)) $nb_holiday=0;
 
-					if ($nb_holiday > 0)
-					{
-						dol_syslog("We update leavefor everybody for type ".$key, LOG_DEBUG);
+				// Update each user counter
+				foreach ($users as $userCounter) {
+					$nbDaysToAdd = $typeleaves[$userCounter['type']]['newByMonth'];
+					if(empty($nbDaysToAdd)) continue;
 
-						$i = 0;
-						while ($i < $nbUser)
-						{
-							$now_holiday = $this->getCPforUser($users[$i]['rowid'], $val['rowid']);
-							$new_solde = $now_holiday + $nb_holiday;
+					dol_syslog("We update leave type id ".$userCounter['type']." for user id ".$userCounter['rowid'], LOG_DEBUG);
 
-							// We add a log for each user
-							$this->addLogCP($user->id, $users[$i]['rowid'], $langs->trans('HolidaysMonthlyUpdate'), $new_solde, $val['rowid']);
+					$nowHoliday = $userCounter['nb_holiday'];
```

Comment:
The property  $userCounter['nb_holiday'];
This was the correct value when there was only 1 counter per user.
But now, there is 1 counter per couple user/type of leave request. The nb of holiday of a user can be get with 
$this->getCPforUser($users[$i]['rowid'], $val['rowid']);
First param is user id, and second is id of type of leave request.